dao/src/test/java/se/fortnox/reactivewizard/db/DaoTransactionsTest.java

```diff
@@ -33,21 +36,18 @@
     private ConnectionProvider connectionProvider = db.getConnectionProvider();
     private DbProxy            dbProxy            = new DbProxy(new DatabaseConfig(), connectionProvider);
     private TestDao            dao                = dbProxy.create(TestDao.class);
-    private DaoTransactions    daoTransactions    = new DaoTransactionsImpl();
+    private DaoTransactions    daoTransactions    = new DaoTransactionsImpl(connectionProvider, dbProxy);
 
     @Test
     public void shouldRunTwoQueriesInOneTransaction() throws SQLException {
         Observable<String> find1 = dao.find();
         Observable<String> find2 = dao.find();
 
-        daoTransactions.createTransaction(find1, find2);
-
-        find1.subscribe();
-
+        Observable<Void> executeTransactionObs = daoTransactions.executeTransaction(find1, find2);
         db.verifyConnectionsUsed(0);
         verify(db.getConnection(), times(0)).prepareStatement(any());
 
-        find2.toBlocking().singleOrDefault(null);
+        executeTransactionObs.toBlocking().subscribe();
```

Comment:
A small possibility of a race condition here since you start verifying stuff before you are certain it has executed.
Perhaps use toBlocking() to be 100% sure